Fix incorrect negation (#258)
kokkos/kokkos-core-wiki: docs/source/API/core/parallel-dispatch/parallel_reduce.rst

  - ``HandleType`` is an ``IntegerType`` if ``ExecPolicy`` is an ``IntegerType`` else it is ``ExecPolicy::member_type``.
* If ``ExecPolicy`` is ``MDRangePolicy`` the ``functor`` has a member function of the form ``operator() (const IntegerType& i0, ... , const IntegerType& iN, ReducerValueType& value) const`` or ``operator() (const WorkTag, const IntegerType& i0, ... , const IntegerType& iN, ReducerValueType& value) const``.

>   - The ``WorkTag`` free form of the operator is used if ``ExecPolicy::work_tag`` is ``void``.

Reviewer: ```suggestion
  - If ``ExecPolicy::work_tag`` is ``void``, the overload without a ``WorkTag`` argument is used.
```

Reviewer 2: The suggested change is easier to read for me.
